File: kmm/composeApp/src/commonMain/kotlin/com/danioliveira/taskmanager/domain/usecase/tasks/DeleteTaskUseCase.kt
package com.danioliveira.taskmanager.domain.usecase.tasks

import com.danioliveira.taskmanager.domain.repository.TaskRepository

/**
 * Use case for deleting tasks.
 *
 * @property taskRepository The repository for task operations
 */
class DeleteTaskUseCase(private val taskRepository: TaskRepository) {
    /**
     * Deletes a task.
     *
     * @param taskId The ID of the task to delete
     * @return Result containing true if the task was deleted successfully, false otherwise
     */
    suspend operator fun invoke(taskId: String): Result<Boolean> {
        return taskRepository.deleteTask(taskId)
    }
} 

File: task_manager_dart/app/lib/features/tasks/domain/usecases/delete_task_usecase.dart
import '../../../../core/utils/result.dart';
import '../../data/repositories/task_repository.dart';

/// Use case for deleting a task
/// Matches KMM's DeleteTaskUseCase
class DeleteTaskUseCase {
  final TaskRepository _repository;

  DeleteTaskUseCase(this._repository);

  Future<Result<void>> call(String taskId) async {
    return await _repository.deleteTask(taskId);
  }
}

